fighters/lucas/src/opff.rs

```diff
@@ -212,15 +212,170 @@ unsafe fn pk_fire_ff(boma: &mut BattleObjectModuleAccessor, stick_y: f32) {
     }
 }
 
+// Offense Up charge Handler
+pub unsafe fn offense_charge(fighter: &mut smash::lua2cpp::L2CFighterCommon, boma: &mut BattleObjectModuleAccessor, situation_kind: i32)  {
+    if(VarModule::is_flag(fighter.object(), vars::lucas::instance::SPECIAL_N_OFFENSE_UP_ACTIVE)) {
+        if fighter.is_status(*FIGHTER_STATUS_KIND_SPECIAL_N) {
+            StatusModule::change_status_request_from_script(fighter.module_accessor, *FIGHTER_LUCAS_STATUS_KIND_SPECIAL_N_FIRE, true);
+        }
```

Comment:
This should be moved to SPECIAL_N's pre status, using set_status_kind_interrupt if the flag is on. Else, call original.
Doing this in opff causes the change_status_request_from_script to run on frame 2 of SPECIAL_N, as status_kind isn't updated until then